src/js/render_wordle.js

```diff
@@ -0,0 +1,214 @@
+        .enter().append("text")
+        .style("font-size", function (d) { return d.size + "px"; })
+        .style("font-family", "Impact")
+        .style("fill", function (d) {
+            return colours(d.text);
+        })
+        .attr("text-anchor", "middle")
+        .attr("transform", function (d) {
+            return "translate(" + [d.x, d.y] + ")rotate(" + d.rotate + ")";
+        })
+        .text(function (d) { return d.text; });
+};
+
+/** Construct the d3 cloud layout object from the supplied Wordle instance
+ * @param {Wordle} instance - The Wordle instance for which the layout is to be constructed
+ * @return {d3Layout} The layout object for the Wordle
+ */
+inverted_wordles.makeLayout = function (instance) {
+    var maxCount = Math.max.apply(null, instance.answerCounts.map(entry => entry.count));
+    var element = instance.element;
+    var laidWords;
+
+    var layout;
+
+    for (var i = 0; i < 10; ++i) {
+        var wordRecs = instance.answerCounts.map(function (entry) {
+            return {text: entry.word, size: instance.conventional ? instance.scale * 120 * entry.count / maxCount :
+                instance.scale * 40 * maxCount / entry.count};
+        });
+        layout = d3.layout.cloud()
+            .size([element.getAttribute("width"), element.getAttribute("height")])
+            .padding(5)
+            .font("Impact")
+            .fontSize(function (d) { return d.size; })
+            // Claims to be event-driven but is actually synchronous
+            .on("end", function (words) {
+                laidWords = words;
+                // Shocking d3 integration model
+            });
+        layout.words(wordRecs);
+        layout.start();
+        if (laidWords.length < wordRecs.length) {
+            instance.scale *= 0.8;
+            console.log("Only laid out " + laidWords.length + " out of " + wordRecs.length + " words, retrying with scale factor " + instance.scale);
+        } else {
+            break;
+        }
+    }
+    inverted_wordles.drawLayout(layout, laidWords, instance);
+    return layout;
+};
+
+/** Given the XHR response, render a wordle instance into its configured selector
+ * @param {WordleInstance} instance - The wordle instance
+ * @param {FetchResponse} response - Fetch response for the answers file from github
+ */
+inverted_wordles.handleResponse = function (instance, response) {
+    response.json().then(function (answersFile) {
+        var answerCounts = inverted_wordles.extractAnswers(answersFile);
+        if (answerCounts.length !== instance.answerCounts.length) {
+            console.log("Updated answerCounts to " + instance.answerCounts.length);
+            instance.answerCounts = answerCounts;
+            inverted_wordles.makeLayout(instance);
+        } else {
+            console.log("No change in answer count");
+        }
+    });
+    // Unfortunately "Etag" is a censored CORS header so we will have to do change detection by brute force
```

Comment:
@amb26, is this still applicable after the client no longer sends cross origin requests?

Shall the old implementation from line 127-153 be removed? If you decided to remove it, it would be useful to keep the comment that explains why the direct fetching from raw.githubusercontent is problematic.

Reply:
Removed with a link to https://issues.fluidproject.org/browse/FLUID-6626